agent/xds/server.go

```diff
@@ -37,23 +38,6 @@ var StatsGauges = []prometheus.GaugeDefinition{
 type ADSStream = envoy_discovery_v3.AggregatedDiscoveryService_StreamAggregatedResourcesServer
 
 const (
-	// Resource types in xDS v3. These are copied from
-	// envoyproxy/go-control-plane/pkg/resource/v3/resource.go since we don't need any of
-	// the rest of that package.
-	apiTypePrefix = "type.googleapis.com/"
-
-	// EndpointType is the TypeURL for Endpoint discovery responses.
-	EndpointType = apiTypePrefix + "envoy.config.endpoint.v3.ClusterLoadAssignment"
-
-	// ClusterType is the TypeURL for Cluster discovery responses.
-	ClusterType = apiTypePrefix + "envoy.config.cluster.v3.Cluster"
-
-	// RouteType is the TypeURL for Route discovery responses.
-	RouteType = apiTypePrefix + "envoy.config.route.v3.RouteConfiguration"
-
-	// ListenerType is the TypeURL for Listener discovery responses.
-	ListenerType = apiTypePrefix + "envoy.config.listener.v3.Listener"
-
 	// PublicListenerName is the name we give the public listener in Envoy config.
 	PublicListenerName = "public_listener"
```

Comment:
For now, I only copy over the minimal code I need based on the POC but moving other constants like this probably makes sense too.

Reply:
IIRC many of these constants exist in the go-control-plane library too. Things changed during the xds v2-to-v3 conversion and some of the library constants were the older deprecated names and some were the new ones and it was unreliable so i switched a lot of things to be local constants instead I could control. Maybe after a future upgrade we could go back to library constants for these things.

Reply:
That makes sense to me. From the perspective of the serverless patcher, it needs to match whatever Consul uses so I think it's still worth putting these here for now.